lib/iris/coords.py

```diff
@@ -625,7 +625,8 @@ def _sanitise_array(self, src, ndmin):
             result = np.require(src, requirements='W')
             # Ensure the array has enough dimensions.
             # NB. Returns the *same object* if result.ndim >= ndmin
-            result = np.array(result, ndmin=ndmin, copy=False)
+            func = ma.array if ma.isMaskedArray(result) else np.array
+            result = func(result, ndmin=ndmin, copy=False)
```

Comment:
As @rcomer points out, this is indeed the nub of the issue...

Reply:
This bit will applies to AuxCoords rather than DimCoords, which was the point of the original issue.
I'll try and remember to edit the squashed commit for that.